app/components/VaultControls/index.js

```diff
@@ -999,8 +1002,29 @@ export default function VaultControls(props) {
                     defaultValue={withdrawTokens[0]}
                     value={selectedWithdrawToken}
                     options={withdrawTokens}
-                    onChange={(newValue) => {
+                    onChange={async (newValue) => {
                       setSelectedWithdrawToken(newValue);
+                      if (
+                        web3Account &&
+                        newValue.address !== currentVaultToken.address
+                      ) {
+                        const approvalStateRes = await fetch(
+                          `https://api.zapper.fi/v1/zap-out/yearn/approval-state?` +
+                            `api_key=96e0cc51-a62e-42ca-acee-910ea7d2a241&sellTokenAddress=` +
+                            `${vaultContract.address.toLowerCase()}&ownerAddress=${web3Account}`,
+                        );
+                        const approvalState = await approvalStateRes.json();
+
+                        if (
+                          approvalState &&
+                          (!approvalState.isApproved ||
+                            approvalState.allowance === '0')
+                        ) {
+                          setWithdrawLabel('Approve');
+                        }
```

Comment:
Think we need 
`} else {
   setWithdrawLabel('Withdraw');
}
`
here also, so that when a user change the zapOut token from a non approved token to an approved one, it displays the _Withdraw_ label. Pease confirm.